channeldb/payments.go

```diff
@@ -22,15 +25,28 @@ var (
 	// paymentStatusBucket is the name of the bucket within the database that
 	// stores the status of a payment indexed by the payment's preimage.
 	paymentStatusBucket = []byte("payment-status")
+
+	// paymentInvoiceIDBucket is a bucket where we map payment hashes to
+	// invoice IDs. This is needed so we can easily lookup the
+	// OutgoingPayment knowing only the payment hash.
+	paymentInvoiceIDBucket = []byte("payment-invoice-id")
```

Comment:
Isn't it better to restructure this part of the database and make payment hash the top level, with status, route and outcome underneath it?

Reply:
route and pid is still in the `router` but the rest is now under a paymenthash bucket!